core/trino-main/src/test/java/io/trino/sql/query/TestSubqueries.java

```diff
@@ -1301,4 +1307,32 @@ public void testCorrelatedUnnestInLateralSubquery()
                         "ON TRUE "))
                 .matches("VALUES 1, 2, 3, 4, 5, null");
     }
+
+    @Test
+    public void testRowSubquery()
```

Comment:
Please add tests for:

- row subquery result being coerced
- a row of `null`s produced in the case when subquery is empty

Reply:
> a row of nulls produced in the case when subquery is empty

Good catch. There was a bug in EnforceSingleRowOperator that I'm fixing.